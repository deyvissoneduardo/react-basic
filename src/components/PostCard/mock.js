export const postCardPropsMock = {
    id: 1,
    title: 'title 01',
    body: 'body 01',
    cover: 'img/img.png'
}