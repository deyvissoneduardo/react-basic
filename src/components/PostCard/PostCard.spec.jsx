import React from "react";
import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";

describe("<PostCard />", () => {
  test("should render PostCard correctly", () => {
    render(<PostCard {...postCardPropsMock} />);

    expect(
      screen.getByRole("img", { name: postCardPropsMock.title })
    ).toHaveAttribute("src", postCardPropsMock.cover);

    expect(
      screen.getByRole("heading", { name: postCardPropsMock.title })
    ).toBeInTheDocument();

    expect(screen.getByText(postCardPropsMock.body)).toBeInTheDocument();
  });

  test("should match snapshot", () => {
    const { container } =   render(<PostCard {...postCardPropsMock} />);

    expect(container.firstChild).toMatchSnapshot()
  });
});
