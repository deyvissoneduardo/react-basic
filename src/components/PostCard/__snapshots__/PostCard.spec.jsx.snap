// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="title 01"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      title 01
    </h1>
    <p>
      body 01
    </p>
  </div>
</div>
`;
